packages/loopring_v3/contracts/amm/libamm/AmmExitProcess.sol

```diff
@@ -111,17 +112,18 @@ library AmmExitProcess
     }
 
     function _burnPoolTokenOnL2(
-        AmmData.Context     memory  ctx,
-        ExchangeData.Block  memory  _block,
-        uint96                      amount,
-        address                     from,
-        uint32                      burnStorageID,
-        bytes               memory  signature
+        AmmData.Context              memory  ctx,
+        bytes                        memory  txsData,
+        uint96                       amount,
+        address                      from,
+        uint32                       burnStorageID,
+        bytes                        memory  signature,
+        TransferTransaction.Transfer memory transfer
         )
         internal
         view
     {
-        TransferTransaction.Transfer memory transfer = _block.readTransfer(ctx.txIdx++);
+        TransferTransaction.readTx(txsData, ctx.txIdx++ * ExchangeData.TX_DATA_AVAILABILITY_SIZE, transfer);
```

Comment:
will use + cheaper than *?

Reply:
In theory ADD costs 3 and MUL costs 5 gas, but in practice the difference is not measurable when used only a few times because of all the other costs that are added on top of it.